pandas/tests/scalar/timedelta/test_constructors.py

```diff
@@ -1,31 +1,78 @@
 from datetime import timedelta
-from itertools import product
+from itertools import (
+    chain,
+    zip_longest,
+)
+import re
 
 import numpy as np
 import pytest
+from toolz.curried import keyfilter
 
 from pandas._libs.tslibs import OutOfBoundsTimedelta
 
 from pandas import (
+    NA,
+    NaT,
     Timedelta,
     offsets,
     to_timedelta,
 )
 
+TD_KWARGS_UNITS = {
+    "weeks": ("w",),
+    "days": ("d", "day", "days"),
+    "hours": ("h", "hr", "hour", "hours"),
+    "minutes": ("m", "t", "min", "minute", "minutes"),
+    "seconds": ("s", "sec", "second", "seconds"),
+    "milliseconds": ("l", "ms", "milli", "millis", "millisecond", "milliseconds"),
+    "microseconds": ("u", "us", "µs", "micro", "micros", "microsecond", "microseconds"),
+    "nanoseconds": ("n", "ns", "nano", "nanos", "nanosecond", "nanoseconds"),
+}
+TD_MAX_PER_KWARG = {
+    "nanoseconds": Timedelta.max.value,
+    "microseconds": Timedelta.max.value // 1_000,
+    "milliseconds": Timedelta.max.value // 1_000_000,
+    "seconds": Timedelta.max.value // 1_000_000_000,
+    "minutes": Timedelta.max.value // (1_000_000_000 * 60),
+    "hours": Timedelta.max.value // (1_000_000_000 * 60 * 60),
+    "days": Timedelta.max.value // (1_000_000_000 * 60 * 60 * 24),
+    "weeks": Timedelta.max.value // (1_000_000_000 * 60 * 60 * 24 * 7),
+}
+TD_MIN_PER_KWARG = {
+    "nanoseconds": Timedelta.min.value,
+    "microseconds": Timedelta.min.value // 1_000,
+    "milliseconds": Timedelta.min.value // 1_000_000,
+    "seconds": Timedelta.min.value // 1_000_000_000,
+    "minutes": Timedelta.min.value // (1_000_000_000 * 60),
+    "hours": Timedelta.min.value // (1_000_000_000 * 60 * 60),
+    "days": Timedelta.min.value // (1_000_000_000 * 60 * 60 * 24),
+    "weeks": Timedelta.min.value // (1_000_000_000 * 60 * 60 * 24 * 7),
+}
+TD_MAX_PER_UNIT = dict(
+    chain.from_iterable(
+        zip_longest(units, (TD_MAX_PER_KWARG[k],), fillvalue=TD_MAX_PER_KWARG[k])
+        for k, units in TD_KWARGS_UNITS.items()
+    )
+)
+TD_MIN_PER_UNIT = dict(
+    chain.from_iterable(
+        zip_longest(units, (TD_MIN_PER_KWARG[k],), fillvalue=TD_MIN_PER_KWARG[k])
+        for k, units in TD_KWARGS_UNITS.items()
+    )
+)
+TD_KWARGS_NP_TD64_UNITS = dict(
+    zip(TD_MAX_PER_KWARG, ("ns", "us", "ms", "s", "m", "h", "D", "W"))
+)
+NP_TD64_MAX_PER_UNIT = dict(
+    zip(("ns", "us", "ms", "s", "m", "h", "D", "W"), TD_MAX_PER_KWARG.values())
+)
+NP_TD64_MIN_PER_UNIT = dict(
+    zip(("ns", "us", "ms", "s", "m", "h", "D", "W"), TD_MIN_PER_KWARG.values())
+)
 
-def test_construct_from_td64_with_unit():
```

Comment:
This any many other tests have just been moved around in the module, so they're better grouped by type.